[#196] view: 진열장 이미지 및 애니메이션 적용

gustn3965 opened the pull request:
## 개요

진열장 이미지 및 애니메이션 적용

![drawer](https://user-images.githubusercontent.com/48749182/141959034-4aa86bdb-75cf-4ec4-ab90-c9892da9dd32.gif)

## 동작 구현 
- 진열장 이미지가 필요한 곳은 `HomeViewController`, `DrawerViewController`,  `SelectingDrawerViewController` 인데, 
`DrawerCoreDataRepostiory` 를 이용하여 데이터가져와서 이미지를 갱신함.
- 새로 진열장을 획득하는 경우, 홈화면에서 애니메이션으로 보여주어야 하는데, 이는 `ViewWillAppear`을 통해, 홈화면으로 재진입시 다시 `DrawerCoreDataRepostiory`를 이용하여 해결 

## 작업 사항
- 필요한 Drawer 이미지 및 애니메이션 파일 추가.
- `DrawerViewController`, `SelectingDrawerViewController`, ` HomeViewController` 이미지 로드하는 로직 추가 
- `HomeViewController`의 `ProfileView`에서 애니메이션뷰가 필요하여, 추가적으로 애니메이션 뷰 삽입.
    - `ProfileView` - GUI에서 미처 반영하지 못한 패딩 값 수정
- 추가적으로 `DrawerRepositoryable` 에 메소드 정의 2개 추가 
    - 진열장 아이템 새로 획득시 이벤트에 대한 처리를 반영하기 위한 메소드들입니다. 
    - 내부적으로, `DrawerCoreDataRepostiory`를 이용하여 진열장 아이탬을 획득하게 되면, 새로 이벤트가 발생했다라고 bool값으로 `keyStore`에 저장합니다. 
- 추가적으로 `DrawerEntity`에 `isNewDrawer` `attribute`를 추가했습니다. 
    - 이것도 새로 획득한 진열장 아이템 여부를 확인하기 위해 추가했습니다. 

- 추가적으로, 실제로 사용자의 동작에 따라, 진열장 아이템을 획득할 수 있도록, `DrawerCoreDataRepostiory`를 적절히 필요한 코드에 삽입했습니다. 
    - 날짜 관련된 진열장 아이템 2개 ( `수학의 정석`, `문구세트` )는 `SceneDelegate`에 삽입했습니다.
    - `드래곤볼`, `VIP`는 `PostRepository`의 `create`, `update` 메소드에 삽입했습니다. 
    - `인의예지상`은 포토카드 구현하면서 삽입할 예정입니다.
    -  장바구니 ( 사고싶다 )는 지원님이 `글쓰기coreData에 반영`하시는 작업할 때, 삽입해주시면 감사하겠습니다! 🙏🏻
        - 사고싶다관련 포스트에서, 구매버튼 누를 때,  아래 코드 삽입해주시면 됩니다!

- 추가적으로 `SettingViewController`의 테스트내용이 많아지다보니 코드가 길어져서 그냥 `SwiftLint`에 except에 추가했습니다. 🙏🏻
```swift   
DrawerCoreDataRepository(coreDataStack: CoreDataStack.shared).updateBasket() 
```
## 테스트 
- 설정화면에서, 날짜 관련 진열장 이외 획득할 수 있도록 구현. 
- 획득하면 애니메이션도 반영되도록 구현.

### Linked Issue

close #196

gustn3965 committed:
[#196] view: 진열장 이미지 및 애니메이션 적용

jwonyLee commented on ThingLog/Coordinator/WriteCoordinator.swift:
```diff
@@ -49,6 +49,9 @@ final class WriteCoordinator: Coordinator {
 
     /// naviagtionController를 dismiss 한다.
     func dismissWriteViewController() {
+        if let home = navigationController.parent as? HomeViewController {
+            
+        }
```
이 부분은 왜 추가되었나요?_?

gustn3965 replied:
아, 여러 시도하면서 안하기로 한 코드였는데 까먹고 삭제하지 못했네요!! 감사합니다!

jwonyLee commented on ThingLog/View/ProfileView.swift:
```diff
@@ -47,11 +47,30 @@ final class ProfileView: UIView {
+        let view: UIView = UIView()
+        view.addSubviews(userBadgeImageView)
+        view.addSubviews(newBadgeAnimationView)
+        view.translatesAutoresizingMaskIntoConstraints = false
```
이 부분은 각각 `view.addSubview`로 고치시거나, `view.addSubviews(userBadgeImageView, newBadgeAnimationView)`로 고치시는 게 좋을 거 같아욤~ 사실 크게 문제가 되는 부분은 아니지만, 약간 이름에 맞게 쓰이는 느낌으로...?

gustn3965 replied:
저도 `addSubViews`를 유용하게 사용하고 있다가 ..ㅎ 저런 실수가 나온 것 같습니다 감사합니다!

jwonyLee reviewed (approved):
LGTM :+1:

nav 바에서 완료 버튼으로 외부로 나갈 때 대표 물건 설정한 내용 반영하는 부분만 수정하시면 될 거 같습니다~! 수정하시고 바로 머지하셔도 될 거 같아요!

gustn3965 committed:
[#196] ref: 지원님 피드백 반영
- 필요없는 코드 제거 및 addViews 한번만 사용
- DrawerViewController에서 네비게이션 바 X버튼 누를 경우 대표진열장 아이템 변경했어도 다시 되돌리도록 변경

gustn3965 commented:
지원님이 피드백 주신 점들 모두 반영했습니다!
